Fix sandbox parameter path for WebKit framework directory

pvollan opened the pull request:
#### 2c92f82e3665ea39d49bb2e4272d16e71117352b
<pre>
Fix sandbox parameter path for WebKit framework directory
<a href="https://bugs.webkit.org/show_bug.cgi?id=243782">https://bugs.webkit.org/show_bug.cgi?id=243782</a>
&lt;rdar://problem/98445723&gt;

Reviewed by Ben Nham.

Currently, the sandbox parameter path for the WebKit framework directory grants sandbox read and extension
access to the whole frameworks directory. We should only grant access to the WebKit framework directory.
This fixes an issue where creation of sandbox extensions fail due to symlinks in this path. When the WebKit
framework is in a non-standard location, we fall back to the old behavior. This is needed for local builds
of WebKit, since the WebCore framework location on disk is a sibling of the WebKit framework location there.

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::populateSandboxInitializationParameters):

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::populateSandboxInitializationParameters):

Canonical link: <a href="https://commits.webkit.org/253777@main">https://commits.webkit.org/253777@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6de3a77adb6578e73192fbaaad0e27694743fba5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31039 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17817 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95823 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29405 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/25712 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79094 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91000 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92568 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/23744 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/73797 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23757 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78733 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79027 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66764 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27145 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/12868 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27082 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13882 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2671 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28762 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/36731 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28705 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33159 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

bnham commented on Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
```diff
@@ -691,7 +691,11 @@ static void populateSandboxInitializationParameters(SandboxInitializationParamet
 
-    sandboxParameters.addPathParameter("WEBKIT2_FRAMEWORK_DIR", [[webKit2Bundle() bundlePath] stringByDeletingLastPathComponent]);
+    String bundlePath = webKit2Bundle().bundlePath;
+    if (!bundlePath.startsWith("/System/Library/Frameworks"_s))
```
Even though you called it out in the commit message, it took me a while to figure out that this is because `WebCore.framework` is a sibling of `WebKit.framework` in local builds rather than within `WebKit.framework/Frameworks` in system builds.

bnham reviewed (approved):
r=me

pvollan committed:
Fix sandbox parameter path for WebKit framework directory
https://bugs.webkit.org/show_bug.cgi?id=243782
<rdar://problem/98445723>

Reviewed by Ben Nham.

Currently, the sandbox parameter path for the WebKit framework directory grants sandbox read and extension
access to the whole frameworks directory. We should only grant access to the WebKit framework directory.
This fixes an issue where creation of sandbox extensions fail due to symlinks in this path. When the WebKit
framework is in a non-standard location, we fall back to the old behavior. This is needed for local builds
of WebKit, since the WebCore framework location on disk is a sibling of the WebKit framework location there.

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::populateSandboxInitializationParameters):

* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::populateSandboxInitializationParameters):

Canonical link: https://commits.webkit.org/253777@main
